### PR TITLE
Nit: remove superfluous version declaration from pom

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Refactoring

### Description

Removes a superfluous version declaration from pom.
No intended functional impact.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-integration-tests/pom.xml [hunk 1]
```diff
@@ -9,12 +9,12 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
     <parent>
         <artifactId>kroxylicious-parent</artifactId>
         <groupId>io.kroxylicious</groupId>
         <version>0.6.0-SNAPSHOT</version>
     </parent>
-    <modelVersion>4.0.0</modelVersion>
```
Comment (SamBarker) [inline kroxylicious-integration-tests/pom.xml:17]: IIRC model version is required by the maven dtd

**Review by SamBarker — APPROVED** [review]
LGTM

File: kroxylicious-integration-tests/pom.xml (hunk 1 above)
Comment (robobario) [reply kroxylicious-integration-tests/pom.xml:17]: it was shuffled higher up, not removed

**Review by robobario — APPROVED** [review]
LGTM

File: kroxylicious-integration-tests/pom.xml (hunk 1 above)
Comment (SamBarker) [reply kroxylicious-integration-tests/pom.xml:17]: Oh yikes I can't read...

**Pull request merged**